lib/simp/cli/commands/environment/new.rb

```diff
@@ -0,0 +1,177 @@
+require 'simp/cli/commands/command'
+require 'simp/cli/environment/omni_env_controller'
+
+# Cli command to create a new Extra/Omni environment
+#
+# TODO: As more `simp environment` sub-commands are added, a lot of this code
+#      could probably be abstracted into a common class or mixin
+class Simp::Cli::Commands::Environment::New < Simp::Cli::Commands::Command
+  # @return [String] description of command
+  def self.description
+    'Create a new SIMP "Extra" (default) or "omni" environment'
+  end
+
+  # Run the command's `--help` strategy
+  def help
+    parse_command_line(['--help'])
+  end
+
+  # Parse command-line options for this simp command
+  # @param args [Array<String>] ARGV-style args array
+  def parse_command_line(args)
+    # TODO: simp cli should read a config file that can override these defaults
+    # these options (preferrable mimicking cmd-line args)
+    default_strategy = :skeleton
+    options = {
+      action:   :create,
+      strategy: :skeleton,
+      types: {
+        puppet: {
+          enabled:    true,
+          strategy:   default_strategy, # :skeleton, :copy
+          puppetfile: false,
+          puppetfile_install: false,
+          deploy:     false,
+          backend:    :directory,
+          environmentpath:  Simp::Cli::Utils.puppet_info[:config]['environmentpath']
+        },
+        secondary: {
+          enabled:  true,
+          strategy: default_strategy,   # :skeleton, :copy, :link
+          backend:  :directory,
+          environmentpath:  Simp::Cli::Utils.puppet_info[:secondary_environment_path]
+        },
+        writable: {
+          enabled:  true,
+          strategy: default_strategy,   # :skeleton, :copy, :link
+          backend:  :directory,
+          environmentpath:  Simp::Cli::Utils.puppet_info[:writable_environment_path]
+        }
+      }
+    }
+
+    opt_parser = OptionParser.new do |opts|
+      opts.banner = '== simp environment new [options]'
+      opts.separator <<-HELP_MSG.gsub(%r{^ {8}}, '')
```

Comment:
that is real nice, will have to remember to steal that

Reply:
:+1: